src/main/java/sirius/biz/storage/layer2/URLBuilder.java

```diff
@@ -173,12 +173,24 @@ public URLBuilder withVariant(String variant) {
         return this;
     }
 
+    /**
+     * Specifies the name of the file.
+     *
+     * @param path the filename to send to the browser
+     * @return the builder itself for fluent method calls
+     */
+    public URLBuilder withFileName(String path) {
+        this.filename = Files.getFilenameAndExtension(path);
+        return this;
+    }
+
     /**
      * Make the URL a download url using the given filename.
      *
      * @param path the filename to send to the browser
      * @return the builder itself for fluent method calls
      */
+    @Deprecated(forRemoval = true)
```

Comment:
Also document `@deprecated` in JavaDoc with a note to use the combination of `withFileName` and `asDownload` instead